### ISSUE TITLE
Sign-up to newsletter

### ISSUE THREAD

**Issue by m-gurung11** [Logic-Legends/Decision-Making-Frmwk#139]
As a user I want to sign up to ACE's newsletter so I can receive correspondence about upcoming projects and plans from ACE.

### Acceptance Criteria

**Scenario 1 - Sign up button**
GIVEN I am using the Decision Making Framework
WHEN I am on the Results screen
I should see a 'Sign-up to newsletter' button at the bottom of the screen

**Scenario 2 - Click on Sign up button**
GIVEN I am on the Results page
WHEN I click the 'Sign-up to newsletter' button
THEN I should get a pop-up with a close (X) icon on the top right corner
AND it should have a text field with hint text above it as 'Please enter your email'

**Scenario 3 - Email validation**
GIVEN I am on the 'Sign up' pop-up
WHEN I enter an email address
THEN it should only accept valid entries

**Valid entries:** The email address should be in the format abc@bcd.com/co.uk

**Scenario 4 - Invalid email**
GIVEN  I have followed scenario 2
WHEN I enter an invalid email address
THEN I should get an error message as 'Please enter in the correct format'

**Scenario 5 - Close**
GIVEN I have followed scenario 2
WHEN I tap/click on the close(X) button
THEN the pop-up should disappear

**Scenario 6 - Email saved in backend table**
GIVEN I have followed scenario 2
WHEN I enter a valid email address
THEN I should be added to the mailing list ( added on the back-end)

**Comment by leilafaez** [comment]
do we need to prevent persisting duplicate email address?

**Comment by leilafaez** [comment]
is there  any default message for sending automatically to user as soon as saving their eamil?

**Comment by m-gurung11** [comment]
@leilafaez yep if you would like to include that functionality then we can have an automatic message saying, 'Thank you for signing-up to ACE's newsletter.'
